### ISSUE TITLE
Importing TensorNetwork is slow

### ISSUE THREAD

**Issue by gecrooks** [google/TensorNetwork#240]
>python -m timeit "import tensornetwork as tn; tn.set_default_backend('numpy'); net = tn.TensorNetwork()"
1 loop, best of 5: 4.12 usec per loop

TensorNetwork is very slow to import. Squinting at the code, it looks like all the backends (numpy, tensorflow, torch, jax) are imported every time, irrespective of the backend that is actually used. In particular, importing TensorFlow is very slow.

**Comment by Thenerdstation** [comment]
I think this PR should've fixed it? https://github.com/google/TensorNetwork/pull/231

We will push a new release soon once https://github.com/google/TensorNetwork/pull/236 is approved

**Comment by gecrooks** [comment]
No, backend_factory imports everything, so I don't think #231 will fix the problem.

**Comment by mganahl** [comment]
actually, it looks like tensorflow is still imported at import tensornetwork

**Comment by Thenerdstation** [comment]
@mganahl How did you test that?

**Comment by mganahl** [comment]
I get and import error if tensorflow isn't installed

**Comment by Thenerdstation** [comment]
Could paste the trace on the other issue?

**Comment by mganahl** [comment]
I  fixed the tensordot2 import in this PR #243 . Apparently importing as a class variable caused
tensorflow to be imported in tensordot2. 

I actually don't see that any of tensorflow, jax or torch get imported at tensorflow import (importing tensorflow in a clean env with only numpy, graphviz, h5py and opt_einsum installed works fine).

**Comment by MichaelMarien** [comment]
@mganahl thanks for fixing. Depending on the outcome of #241 I can change the test to fix this behaviour (core reason is that I do not cleanup the import of `tensornetwork.backends.tensorflow` in the tests.
